.github/workflows/risc0-ethereum.yml

```
@@ -0,0 +1,146 @@
+        run: exit 1
+
+  test:
+    if: ${{ needs.changes.outputs.test == 'true' || github.event_name == 'workflow_dispatch'}}
+    needs: changes
+    runs-on: [self-hosted, prod, Linux, cpu, docker]
+    env:
+      RUST_BACKTRACE: full
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
+      - run: "git checkout -f $(git -c user.name=x -c user.email=x@x commit-tree $(git hash-object -t tree /dev/null) < /dev/null) || :"
+      - uses: actions/checkout@v4
+        with:
+          path: 'risc0'
+      - uses: ./risc0/.github/actions/rustup
+      - uses: ./risc0/.github/actions/sccache
+        with:
+          key: Linux-default
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install python dependencies
+        run: python -m pip install --upgrade toml
+      - name: Install risczero toolchain
+        run: |
+          cargo install --force --path risc0/cargo-risczero
+          cargo risczero install --version $RISC0_TOOLCHAIN_VERSION
+        working-directory: risc0
+      - uses: risc0/foundry-toolchain@2fe7e70b520f62368a0e3c464f997df07ede420f
+        with:
+          version: nightly-09fe3e041369a816365a020f715ad6f94dbce9f2
+      - name: Checkout risc0-ethereum
+        uses: actions/checkout@v4
+        with:
+          repository: 'risc0/risc0-ethereum'
+          token: ${{ secrets.GITHUB_TOKEN }}
+          submodules: recursive
+          path: 'risc0-ethereum'
+          ref: ${{ github.event.inputs.risc0-ethereum-branch || 'main' }}
+      - name: Local patch on risc0-ethereum
+        run: python ../risc0/.github/cargo_local_patch.py Cargo.toml
+        working-directory: risc0-ethereum
+      - name: Bootstrap-groth16 test receipt
+        run: cargo xtask bootstrap-groth16 --test-receipt-only
+        working-directory: risc0
+      - name: build risc0-ethereum
+        run: cargo test --workspace --no-run
+        working-directory: risc0-ethereum
+      - name: test risc0-ethereum
+        run: cargo test --workspace
+        working-directory: risc0-ethereum
+      - name: forge test on risc0-ethereum
+        run: forge test -vvv
+        working-directory: risc0-ethereum/contracts
+      - name: Checkout foundry-template
+        uses: actions/checkout@v4
+        with:
+          repository: 'risc0/bonsai-foundry-template'
+          token: ${{ secrets.GITHUB_TOKEN }}
+          submodules: recursive
+          path: 'bonsai-foundry-template'
+          ref: ${{ github.event.inputs.bonsai-foundry-template-branch || 'main' }}
+      - name: Local patch on bonsai-foundry-template
+        run: |
+          python ../risc0/.github/cargo_local_patch.py Cargo.toml
+          python ../risc0/.github/cargo_local_patch.py methods/guest/Cargo.toml
+          python ../risc0/.github/cargo_local_patch.py lib/risc0-ethereum/Cargo.toml
```

Maybe we could have the script search recursively for `Cargo.toml` files instead of maintaining these three lines. It just seems easy to miss a place where a substitute should have been made. Also, there is another workspace in `risc0-ethereum`, which is the guest used to test the relay, so I guess that's an example of missing a `Cargo.toml` file.
```suggestion
          python ../risc0/.github/cargo_local_patch.py Cargo.toml
          python ../risc0/.github/cargo_local_patch.py methods/guest/Cargo.toml
          python ../risc0/.github/cargo_local_patch.py lib/risc0-ethereum/Cargo.toml
```